compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -138,15 +138,31 @@ private Optional<Path> modulePath(ModuleId moduleId) {
         return Optional.empty();
     }
 
+    private Optional<Path> generatedModulePath(ModuleId moduleId) {
+        if (currentPackage().moduleIds().contains(moduleId)) {
+            if (currentPackage().getDefaultModule().moduleId() == moduleId) {
+                return Optional.of(sourceRoot);
```

Comment:
Shouldn't this return `sourceRoot.resolve(ProjectConstants.GENERATED_MODULES_ROOT)`?

Reply:
yes. I'll correct.